src/routes/Messages/ThreadList/ThreadList.js

```diff
@@ -25,15 +26,21 @@ export default class ThreadList extends Component {
       currentUser,
       threadsPending,
       threads,
+      onCloseURL,
       threadSearch,
       onScrollBottom,
+      messagesOpen,
+      toggleMessages,
       match: { params: { messageThreadId } },
       className
     } = this.props
 
     return <div styleName='thread-list' className={className}>
       <div styleName='header'>
-        <Link to='/t/new'><Button label='New Message' styleName='new-message' /></Link>
+        <div styleName='closeMessages'>
+          <CloseMessages onCloseURL={onCloseURL} />
+        </div>
+        <Link to='/t/new' onClick={toggleMessages}><Button label='New Message' styleName='new-message' /></Link>
```

Comment:
this seems weird to me, a Button inside a Link that has both a url destination and an onClick handler. lets look at this together, i want to understand it